apps/dcos_l4lb/src/dcos_l4lb_ipvs_mgr.erl

```diff
@@ -133,30 +134,48 @@ init([]) ->
     {ok, #state{netns = #{host => Pid}, family = Family}}.
 
 handle_call({get_services, Namespace}, _From, State) ->
-    Reply = handle_get_services(Namespace, State),
+    Reply = handle_call_track_time(fun handle_get_services/2, [Namespace, State]),
     {reply, Reply, State};
 handle_call({add_service, IP, Port, Protocol, Namespace}, _From, State) ->
-    Reply = handle_add_service(IP, Port, Protocol, Namespace, State),
+    Reply = handle_call_track_time(fun handle_add_service/5, [IP, Port, Protocol, Namespace, State]),
     {reply, Reply, State};
 handle_call({remove_service, IP, Port, Protocol, Namespace}, _From, State) ->
-    Reply = handle_remove_service(IP, Port, Protocol, Namespace, State),
+    Reply = handle_call_track_time(fun handle_remove_service/5, [IP, Port, Protocol, Namespace, State]),
     {reply, Reply, State};
 handle_call({get_dests, Service, Namespace}, _From, State) ->
-    Reply = handle_get_dests(Service, Namespace, State),
+    Reply = handle_call_track_time(
+        fun handle_get_dests/3, [Service, Namespace, State]),
     {reply, Reply, State};
 handle_call({add_dest, ServiceIP, ServicePort, DestIP, DestPort, Protocol, Namespace}, _From, State) ->
-    Reply = handle_add_dest(ServiceIP, ServicePort, DestIP, DestPort, Protocol, Namespace, State),
+    Reply = handle_call_track_time(
+        fun handle_add_dest/7,
+        [ServiceIP, ServicePort, DestIP, DestPort, Protocol, Namespace, State]),
     {reply, Reply, State};
 handle_call({remove_dest, ServiceIP, ServicePort, DestIP, DestPort, Protocol, Namespace}, _From, State) ->
-    Reply = handle_remove_dest(ServiceIP, ServicePort, DestIP, DestPort, Protocol, Namespace, State),
+    Reply = handle_call_track_time(
+        fun handle_remove_dest/7,
+        [ServiceIP, ServicePort, DestIP, DestPort, Protocol, Namespace, State]),
     {reply, Reply, State};
 handle_call({add_netns, UpdateValue}, _From, State0) ->
-    {Reply, State1} = handle_add_netns(UpdateValue, State0),
+    {Reply, State1} = handle_call_track_time(fun handle_add_netns/2, [UpdateValue, State0]),
     {reply, Reply, State1};
 handle_call({remove_netns, UpdateValue}, _From, State0) ->
-    {Reply, State1} = handle_remove_netns(UpdateValue, State0),
+    {Reply, State1} = handle_call_track_time(
+        fun handle_remove_netns/2, [UpdateValue, State0]),
     {reply, Reply, State1}.
 
+handle_call_track_time(Fun, Args) ->
+    Begin = erlang:monotonic_time(),
+    Return = erlang:apply(Fun, Args),
+    try
```

Comment:
I can't really say whether it was hard, but I was not able to do it without adding the try. If you could spare some time to look at it and give me some hints that would be awesome. Otherwise, I can give it another shot once I have managed to finally test all the metrics.

Reply:
I'm planning to take a look at all code related to tests after all other questions are resolved.